tasks/human_eval_139.rs

```diff
@@ -5,14 +5,151 @@ HumanEval/139
 /*
 ### VERUS BEGIN
 */
+use vstd::arithmetic::mul::*;
 use vstd::prelude::*;
 
 verus! {
 
-// TODO: Put your solution (the specification, implementation, and proof) to the task here
+// specification
+pub closed spec fn brazilian_factorial_helper(
+    start: int,
+    end: int,
+    fact_i: int,
+    special_fact: int,
+) -> int
+    decreases (end - start),
+{
+    if (end - start <= 0) {
+        special_fact
+    } else {
+        brazilian_factorial_helper(start + 1, end, fact_i * start, special_fact * (fact_i * start))
+    }
+}
+
+pub closed spec fn brazilian_factorial(n: int) -> int {
+    brazilian_factorial_helper(1int, n + 1, 1, 1)
+}
+
+// implementation
+proof fn lemma_bf_helper_is_increasing(start: nat, end: nat, fact_i: nat, special_fact: nat)
+    requires
+        start >= 1,
+        fact_i >= 1,
+        special_fact >= 1,
+    ensures
+        special_fact <= brazilian_factorial_helper(
+            start as int,
+            end as int,
+            fact_i as int,
+            special_fact as int,
+        ),
+    decreases (end - start),
+{
+    if (end - start <= 0) {
+    } else {
+        assert(fact_i * start >= 1) by { lemma_mul_strictly_positive(fact_i as int, start as int) };
+        assert((special_fact * (fact_i * start)) >= special_fact) by {
+            lemma_mul_ordering(special_fact as int, (fact_i * start) as int)
+        };
+        assert(brazilian_factorial_helper(
+            (start + 1) as int,
+            end as int,
+            (fact_i * start) as int,
+            (special_fact * (fact_i * start)) as int,
+        ) >= (special_fact * (fact_i * start))) by {
+            lemma_bf_helper_is_increasing(
+                start + 1,
+                end,
+                fact_i * start,
+                special_fact * (fact_i * start),
+            )
+        };
+    }
+}
+
+pub fn brazilian_factorial_impl(n: u32) -> (bf: u32)
+    requires
+        0 < n < 4294967295,
```

Comment:
Can you replace 4294967295 with `u32::MAX` to make the intent clearer?

I also suspect that the next requirement is a strictly stronger requirement, no?

I'm also a bit reluctant to have a requirement that asks that you already know something about the thing you're about to compute.  Could we instead use a strategy similar to [this version](https://github.com/secure-foundations/human-eval-verus/blob/24726b00c443015ac68ee999498b49dba64422c8/tasks/human_eval_055a.rs#L42) of `fib`?